src/misceval.js

```diff
@@ -624,12 +631,22 @@ Sk.misceval.isTrue = function (x) {
     if (x.constructor === Sk.builtin.float_) {
         return x.v !== 0;
     }
-    if (x["__nonzero__"]) {
-        ret = Sk.misceval.callsimArray(x["__nonzero__"], [x]);
-        if (!Sk.builtin.checkInt(ret)) {
-            throw new Sk.builtin.TypeError("__nonzero__ should return an int");
+    if (Sk.__future__.python3) {
+        if (x["__bool__"]) {
```

Comment:
why do you use `["__bool__"]` here instead of `__bool__`

Reply:
Because the access to ```["__len__"]``` right below this was already there, so we maintained consistency with the existing code.  I can trivially switch them all over if you'd like...

Reply:
If closure is happy with it I think that would make sense.

Reply:
Just to be clear, do you think I should make them all like ```x["__bool__"]``` or ```x.__bool__```?  I think you mean the latter...